Update PETSc

milljm opened the pull request:
Updating PETSc default to 3.8.3. Bumping 3.7.6 to Alternate
version. Thus removing PETSc 3.6.4.

Updating SLEPc default/alt to match PETSc's default/alt update.

Commit:
Update PETSc

Updating PETSc default to 3.8.3. Bumping 3.7.6 to Alternate
version. Thus removing PETSc 3.6.4.

Updating SLEPc default/alt to match PETSc's default/alt update.
update patch code due to version switch

Fix the patching logic due to PETSc version identifier
swap.

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/168825/) on d635dfd wanted to post the following:

View the site [here](http://mooseframework.inl.gov/docs/PRs/144/site)

This comment will be updated on new commits.

Commit:
Change of plans

We want PETSc 3.8.3 to be the Alternate and 3.7.6 to remain
the default, until folks can fix there stuff with 3.8.3.
